src/server/pfs/server/pfsserver.proto

```diff
@@ -1,20 +1,40 @@
 syntax = "proto3";
 
-package pfsserver;
+package fileset;
 option go_package = "github.com/pachyderm/pachyderm/v2/src/server/pfs/server";
 
-message CompactionTask {
+message ShardTask {
   int64 index = 1;
   repeated string inputs = 2;
-  PathRange range = 3;
 }
 
-message CompactionTaskResult {
+message ShardTaskResult {
   int64 index = 1;
-  string id = 2;
+  repeated CompactTask compact_tasks = 2;
 }
 
 message PathRange {
   string lower = 1;
   string upper = 2;
 }
+
+message CompactTask {
+  int64 index = 1;
```

Comment:
Why do all of these have an index field?

Reply:
The order we collect the results in doesn't correspond with the order the tasks were created, so we need this to map back to the ordering of the tasks.